tests/Aspire.Hosting.Tests/Elasticsearch/AddElasticsearchTests.cs

```diff
@@ -0,0 +1,238 @@
+        Assert.Equal(ElasticsearchContainerImageTags.Tag, containerAnnotation.Tag);
+        Assert.Equal(ElasticsearchContainerImageTags.Image, containerAnnotation.Image);
+        Assert.Equal(ElasticsearchContainerImageTags.Registry, containerAnnotation.Registry);
+
+        var config = await EnvironmentVariableEvaluator.GetEnvironmentVariablesAsync(containerResource);
+
+        Assert.Collection(config,
+            env =>
+            {
+                Assert.Equal("discovery.type", env.Key);
+                Assert.Equal("single-node", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("xpack.security.enabled", env.Key);
+                Assert.Equal("true", env.Value);
+            },
+            env =>
+            {
+                Assert.Equal("ELASTIC_PASSWORD", env.Key);
+                Assert.Equal("pass", env.Value);
+            });
+    }
+
+    [Fact]
+    public async Task ElasticsearchCreatesConnectionString()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        var elasticsearch = appBuilder
+            .AddElasticsearch("elasticsearch")
+            .WithEndpoint("http", e => e.AllocatedEndpoint = new AllocatedEndpoint(e, "localhost", 27020));
+
+        using var app = appBuilder.Build();
+
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+
+        var connectionStringResource = Assert.Single(appModel.Resources.OfType<ElasticsearchResource>()) as IResourceWithConnectionString;
+        var connectionString = await connectionStringResource.GetConnectionStringAsync();
+
+        Assert.Equal($"http://{elasticsearch.Resource.UserNameParameter.Value}:{elasticsearch.Resource.PasswordParameter.Value}@localhost:27020", connectionString);
+        Assert.Equal("http://{elasticsearch-username.value}:{elasticsearch-password.value}@{elasticsearch.bindings.http.host}:{elasticsearch.bindings.http.port}", connectionStringResource.ConnectionStringExpression.ValueExpression);
+    }
+
+    [Fact]
+    public async Task VerifyManifestWithDefaultsAddsAnnotationMetadata()
+    {
+        using var appBuilder = TestDistributedApplicationBuilder.Create();
+
+        var kafka = appBuilder.AddElasticsearch("elasticsearch");
+
+        var manifest = await ManifestUtils.GetManifest(kafka.Resource);
+
+        var expectedManifest = $$"""
+            {
+              "type": "container.v0",
+              "connectionString": "http://{elasticsearch-username.value}:{elasticsearch-password.value}@{elasticsearch.bindings.http.host}:{elasticsearch.bindings.http.port}",
+              "image": "{{ElasticsearchContainerImageTags.Registry}}/{{ElasticsearchContainerImageTags.Image}}:{{ElasticsearchContainerImageTags.Tag}}",
+              "env": {
+                "discovery.type": "single-node",
+                "xpack.security.enabled": "true",
+                "ELASTIC_PASSWORD": "{elasticsearch-password.value}"
```

Comment:
What is the deal with the username in Elasticsearch? We provide it on the connection string but it doesn't seem to be passed into the container for configuration ... is it ignored?

Reply:
changing Elasticsearch's built-in username (like `elastic`) is impossible, but clients like `Elastic.Clients.Elasticsearch` have basic authentication using a username and password (besides using credentials in the URL).
For example :

`var settings = new ElasticsearchClientSettings(new Uri("http://localhost:9200")).Authentication(new BasicAuthentication("<USERNAME>", "<PASSWORD>"));`

so, I put the username parameter in the resource. we could remove this property from the resource and mention in the document that the username is `elastic`.

Any suggestion?

@mitchdenny , @davidfowl

Reply:
If the username is always hardcoded (truly) to elastic then we can make it constant. No need to generate a parameter for something that could only be the one value (and people could type in the wrong thing leading to issues).